### ISSUE TITLE
No warnings for field initializer with implicit constructor

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#31139]
A warning is reported for `A.x` but no warning is reported for `B.y`.
```c#
class A
{
    object x = F(null); // warning
    A() { }
    static object F(object o) => o;
}
class B
{
    object y = F(null); // no warning
    static object F(object o) => o;
}
```
